### PR TITLE
resolve Issues/13

### PR DESCRIPTION
**Pull request by youngkiu**
https://github.com/cpederkoff/stl-to-voxel/issues/13

1. Improved performance by parallel processing with ray library.
2. The number of padding pixels can be set as a command argument.
3. The padding pixel was reflected in advance when allocating volume memory.
4. When the output files are png and svx, performance is improved.
5. The volume index was maintained in the order of z, y, x.
6. In the same style as linesToVoxels(), meshToPlane() was newly created.
7. Fundamentally handled the case where the number of lines is not even.

### CODE REVIEW

File: slice.py [hunk 1]
```diff
@@ -92,14 +144,24 @@ def calculateScaleAndShift(mesh, resolution):
 
 
 def scaleAndShiftMesh(mesh, scale, shift):
+    adjusted_mesh = []
     for tri in mesh:
```
Comment (cpederkoff) [inline slice.py:148]: I think this for loop can be replaced with 
`adjusted_mesh = (mesh + shift) * scale`

File: stltovoxel.py [hunk 2]
```diff
@@ -68,40 +43,37 @@ def exportPngs(voxels, bounding_box, outputFilePath):
 
     size = str(len(str(bounding_box[2]))+1)
     for height in range(bounding_box[2]):
-        img = Image.new('L', (bounding_box[0], bounding_box[1]), 'black')  # create a new black image
-        pixels = img.load()
-        arrayToWhiteGreyscalePixel(voxels[height], pixels)
+        print('export png %d/%d' % (height, bounding_box[2]))
+        img = Image.fromarray(voxels[height])
```
Comment (cpederkoff) [inline stltovoxel.py:47]: nice

File: stltovoxel.py [hunk 3]
```diff
@@ -126,6 +98,7 @@ def file_choices(choices, fname):
         type=lambda s: file_choices(('.png', '.xyz', '.svx'), s),
         help='path to output files. The export data type is chosen by file extension. Possible are .png, .xyz and .svx')
     parser.add_argument('resolution', nargs='?', type=int, default=100, help='number of voxels in both directions')
+    parser.add_argument('pad', nargs='?', type=int, default=1, help='number of padding bits')
```
Comment (cpederkoff) [inline stltovoxel.py:101]: number of padding pixels

File: slice.py [hunk 4]
```diff
@@ -14,55 +63,50 @@ def linearInterpolation(p1, p2, distance):
     :param distance: Between 0 and 1, Lower numbers return points closer to p1.
     :return: A point on the line between p1 and p2
     '''
-    slopex = (p1[0] - p2[0])
```
Comment (cpederkoff) [inline slice.py:17]: I think the body of this function can be replaced with `return p1 * (1-distance) + p2 * distance`

**Review by cpederkoff — CHANGES_REQUESTED** [review]
A few small changes. Otherwise looks great.

File: stltovoxel.py (hunk 3 above)
Comment (youngkiu) [reply stltovoxel.py]: The pixel seems to be a more accurate representation.
Commit: 7fff41e888b51d76861a9453ec25fe2abe3b1f89

File: slice.py (hunk 1 above)
Comment (youngkiu) [reply slice.py:148]: https://github.com/youngkiu/stl-to-voxel/blob/mesh_numpy/slice.py
I am preparing a new PR, can I review it there?

File: stltovoxel.py (hunk 2 above)
Comment (youngkiu) [reply stltovoxel.py:47]: A great time reduction has been achieved.

Comment (youngkiu) [reply stltovoxel.py:47]: https://github.com/cpederkoff/stl-to-voxel/issues/13#issuecomment-838398666
Commit: aeda356, 527 sec --> Commit: b175859, 73 sec

File: slice.py (hunk 4 above)
Comment (youngkiu) [reply slice.py]: https://github.com/youngkiu/stl-to-voxel/blob/mesh_numpy/slice.py#L71
I am preparing a new PR, can I review it there?

**Review by cpederkoff — APPROVED** [review]

**Pull request merged**